Cyotek.Windows.Forms.ImageBox/ImageBox.cs

```diff
@@ -4062,21 +4098,63 @@ protected override void OnMouseWheel(MouseEventArgs e)
     {
       base.OnMouseWheel(e);
 
-      if (this.AllowZoom && this.SizeMode == ImageBoxSizeMode.Normal)
+      void DoZoom()
       {
-        int spins;
+        if (this.AllowZoom && this.SizeMode == ImageBoxSizeMode.Normal)
+        {
+          int spins;
 
-        // The MouseWheel event can contain multiple "spins" of the wheel so we need to adjust accordingly
-        spins = Math.Abs(e.Delta / SystemInformation.MouseWheelScrollDelta);
+          // The MouseWheel event can contain multiple "spins" of the wheel so we need to adjust accordingly
+          spins = Math.Abs(e.Delta / SystemInformation.MouseWheelScrollDelta);
 
-        // TODO: Really should update the source method to handle multiple increments rather than calling it multiple times
-        for (int i = 0; i < spins; i++)
+          // TODO: Really should update the source method to handle multiple increments rather than calling it multiple times
+          for (int i = 0; i < spins; i++)
+          {
+            this.ProcessMouseZoom(e.Delta > 0, e.Location);
+          }
+        }
+      }
+
+      if (MouseWheelMode == ImageBoxMouseWheelMode.Zoom)
+      {
+        DoZoom();
+      }
+      else if (MouseWheelMode == ImageBoxMouseWheelMode.ScrollAndZoom)
+      {
+        if (ModifierKeys == Keys.Control)
+        {
+          DoZoom();
+        }
+        else if (VerticalScroll.Visible && VerticalScroll.Enabled && ModifierKeys == Keys.None)
         {
-          this.ProcessMouseZoom(e.Delta > 0, e.Location);
+          int scrollDelta = SystemInformation.MouseWheelScrollLines * VerticalScroll.SmallChange;
+          ScrollTo(HorizontalScroll.Value, VerticalScroll.Value + ((e.Delta > 0) ? -scrollDelta : scrollDelta));
         }
+        else if (HorizontalScroll.Visible && HorizontalScroll.Enabled && ModifierKeys == Keys.Shift)
+        {
+          int scrollDelta = SystemInformation.MouseWheelScrollLines * HorizontalScroll.SmallChange;
+          ScrollTo(HorizontalScroll.Value + ((e.Delta > 0) ? -scrollDelta : scrollDelta), VerticalScroll.Value);
+        }
+      }
+      else
+      {
+        throw new NotImplementedException("Mouse wheel mode not implemented");
```

Comment:
Validation of the enum value should be made when the property is set, when it can be dealt with. Throwing an exception here would probably be catastrophic for the user. I have encountered similar issues in the past when the `Paint` methods crash and while it didn't kill the application, it did kill the control.

Please remove this else block completely so that nothing happens if an illegal value has been set.

Reply:
Sounds plausible. After adding the "None" mode, this makes even more sense.